builtin/providers/aws/resource_aws_instance.go

```diff
@@ -475,6 +475,9 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("subnet_id", instance.SubnetId)
 	}
 	d.Set("ebs_optimized", instance.EbsOptimized)
+	if instance.SubnetId != nil && *instance.SubnetId != "" {
+		d.Set("source_dest_check", instance.SourceDestCheck)
```

Comment:
@phinze Since the attribute only applies in VPCs, I figured it was better to only set it for those instances and keep it out of the state file for non-VPC instances.